File: src/main/java/no/nav/fo/forenkletdeploy/service/VeraDeployService.java
package no.nav.fo.forenkletdeploy.service;

import no.nav.fo.forenkletdeploy.domain.VeraDeploy;
import no.nav.fo.forenkletdeploy.domain.VeraDeploy.VeraDeploys;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.cache.annotation.Cacheable;
import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.List;

import static no.nav.fo.forenkletdeploy.util.Utils.withClient;

@Component
public class VeraDeployService {
    private static Logger LOG = LoggerFactory.getLogger(VeraDeployService.class.getName());
    @Cacheable("veradeploys")
    public List<VeraDeploy> getVeraDeploys() {
        try {
            return withClient("https://vera.adeo.no/api/v1/deploylog?onlyLatest=true&filterUndeployed=true")
                    .request()
                    .get(VeraDeploys.class);
        } catch(Exception e) {
            LOG.error("Kunne ikke hente deploys fra vera", e);
            return new ArrayList<>();
        }
    }
}

File: src/main/kotlin/no/nav/fo/forenkletdeploy/service/VeraDeployService.kt
package no.nav.fo.forenkletdeploy.service

import no.nav.fo.forenkletdeploy.VeraDeploy
import no.nav.fo.forenkletdeploy.consumer.getVeraConsumer
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import java.util.*
import javax.inject.Inject

@Service
class VeraDeployService @Inject
constructor(
        val teamService: TeamService
) {
    val veraConsumer = getVeraConsumer()
    private val LOG = LoggerFactory.getLogger(VeraDeployService::class.java)
    private val gyldigeMiljoer = Arrays.asList("p", "q0", "q6", "t6")

    fun getDeploysForTeam(teamId: String): List<VeraDeploy> =
            teamService.getTeam(teamId).getApplicationConfigs()
                    .flatMap { getDeploysForApp(it.name) }

    fun getDeploysForApp(app: String): List<VeraDeploy> =
            try {
                veraConsumer.getDeploysForApp(app)
                        .filter { gyldigeMiljoer.contains(it.environment) }
            } catch (e: Throwable) {
                LOG.error("Kunne ikke hente deploys for $app", e)
                emptyList()
            }
}
